listenstore/listenstore/listenstore.py

```diff
@@ -188,24 +193,70 @@ def get_playing_now(self, user_id):
 
 REDIS_INFLUX_USER_LISTEN_COUNT = "ls.listencount." # append username
 class InfluxListenStore(ListenStore):
+
+    # variables which define how much time to cache listen count in redis
+    LISTENCOUNT_CACHE_TIME_LOW = 15 * 60 # in seconds. 15 minutes
+    LISTENCOUNT_CACHE_TIME_MEDIUM = 60 * 60 # in seconds. 1 hour
+    LISTENCOUNT_CACHE_TIME_HIGH = 24 * 60 * 60 # in seconds. 1 day
+    LOW_LISTEN_THRESHOLD = 1000
+    MEDIUM_LISTEN_THRESHOLD = 10000
+
     def __init__(self, conf):
         ListenStore.__init__(self, conf)
         self.redis = Redis(conf['REDIS_HOST'])
         self.influx = InfluxDBClient(host=conf['INFLUX_HOST'], port=conf['INFLUX_PORT'], database=conf['INFLUX_DB_NAME'])
 
 
-    def _get_num_listens_for_user(self, user_name):
-        count = self.redis.get(REDIS_INFLUX_USER_LISTEN_COUNT + user_hash(user_name))
-        if count:
-            return int(count)
+    def get_listen_count_for_user(self, user_name, need_exact = False):
+        """ Returns total listen count for user with user_name passed.
 
-        results = self.influx.query("""SELECT count(*) 
-                                         FROM listen 
-                                        WHERE user_name = '%s'""" % (user_name, from_ts, to_ts, limit))
-        result = results.get_points(measurement='listen')
-#        count = self.redis.get(REDIS_INFLUX_USER_LISTEN_COUNT + user_hash(user_name))
+            Args:
+                user_name: user name of user whose listen count is to be found
+                need_exact: signifies if cached value in redis will suffice or if exact value is needed
+        """
 
-        return 0
+        if not need_exact:
+            # check if the user's listen count is already in redis
+            # if already present return it directly instead of calculating it again
+            count = self.redis.get(REDIS_INFLUX_USER_LISTEN_COUNT + user_name)
+            if count:
+                return int(count)
+
+        try:
+            results = self.influx.query("""SELECT count(*)
+                                             FROM listen
+                                            WHERE user_name = '%s'""" % (user_name))
```

Comment:
Sorry, I think this might have been waiting on me to respond to.
You're right, I had a look at the docs and some of the implementation and I really can't find much around query parameters. I'm really worried about us doing this manually instead of getting a tool to do it for us. Any place where users can provide data, we're just asking for trouble if we try and sanitize input before we run a query.
I wonder if since influxql is quite similar to sql, could we use something like http://initd.org/psycopg/docs/cursor.html#cursor.mogrify to escape for us?  It does mean that we need a postgres cursor - perhaps the code is simple enough that we could copy it into a lib and then change all places we use query params in influx to use this wrapper.